plugins/modules/mso_schema_template_contract_filter.py

```diff
@@ -199,6 +224,13 @@ def main():
         filter_template = filter_template.replace(' ', '')
     filter_schema = module.params.get('filter_schema')
     filter_type = module.params.get('filter_type')
+    qos_level = module.params.get('qos_level')
+    action = module.params.get('action')
+    priority = module.params.get('priority')
+    if priority is not None:
+        priority = priority.lower()
+        priority = priority.replace(' ', '_')
```

Comment:
Why not put the choices directly in the priority_map and not try to lower case / _ them?